EventProc/src/HcalDigiProducer.cxx

```diff
@@ -128,76 +145,151 @@ namespace ldmx {
                 numSigHits_side_tb++;
             else if( tempID.getSection() == HcalSection::LEFT || tempID.getSection() == HcalSection::RIGHT )
                 numSigHits_side_lr++;
-	    else 
-	        std::cout << "WARNING [HcalDigiProducer::produce]: HcalSection is not known" << std::endl;
-	    
-            hcalLayerPEs[detIDraw] = random_->Poisson(meanPE+meanNoise_);
-            //hcalLayerPEs[detIDraw] += random_->Gaus(meanNoise_);
-
-            if (verbose_) {
-                detID_->setRawValue(detIDraw);
-                detID_->unpack();
-                int layer = detID_->getFieldValue("layer");
-                int subsection = detID_->getFieldValue("section");
-                int strip = detID_->getFieldValue("strip");
-
-                std::cout << "detID: " << detIDraw << std::endl;
-                std::cout << "Layer: " << layer << std::endl;
-                std::cout << "Subsection: " << subsection << std::endl;
-                std::cout << "Strip: " << strip << std::endl;
-                std::cout << "Edep: " << hcaldetIDEdep[detIDraw] << std::endl;
-                std::cout << "numPEs: " << hcalLayerPEs[detIDraw] << std::endl;
-                std::cout << "time: " << hcaldetIDTime[detIDraw] << std::endl;
-                std::cout << "z: " << hcalZpos[detIDraw] << std::endl;
-            }        // end verbose
+	        else std::cout << "WARNING [HcalDigiProducer::produce]: HcalSection is not known" << std::endl;
 
             // need to add in a weighting factor eventually, so keep it that way to make sure
             // we don't forget about it
             double energy = depEnergy; 
 
-            if( hcalLayerPEs[detIDraw] > readoutThreshold_ ){
+            // quantize/smear the position
+            HcalID *curDetId = new HcalID();
+            curDetId->setRawValue(detIDraw);
+            curDetId->unpack();
+            int cur_subsection = curDetId->getFieldValue("section");            
+            int cur_layer      = curDetId->getFieldValue("layer");
+            int cur_strip      = curDetId->getFieldValue("strip");
+            float cur_xpos, cur_ypos; 
+
+            if (cur_subsection != 0){ // for sidecal don't worry about attenuation because it's single readout
+                hcalLayerPEs[detIDraw] = random_->Poisson(meanPE+meanNoise_);
+                hcalLayerMinPEs[detIDraw] = hcalLayerPEs[detIDraw];
+            }
+            if (cur_subsection == 0){// get PEs with attentuation
```

Comment:
Should we make the attenuation a parameter?

Reply:
Yes that's a good idea.  Also another parameter for the time smear resolution?